daemon/cluster/services.go

```diff
@@ -134,6 +138,35 @@ func (c *Cluster) CreateService(s types.ServiceSpec, encodedAuth string, queryRe
 
 		switch serviceSpec.Task.Runtime.(type) {
 		// handle other runtimes here
+		case *swarmapi.TaskSpec_Generic:
+			switch serviceSpec.Task.GetGeneric().Kind {
+			case string(types.RuntimePlugin):
+				if s.TaskTemplate.PluginSpec == nil {
+					return errors.New("plugin spec must be set")
+				}
+				p := runtime.GenericPluginPayload{
+					Spec: *s.TaskTemplate.PluginSpec,
+				}
+				if encodedAuth != "" {
+					p.PullOptions = runtime.PluginPullOptions{RegistryAuth: encodedAuth}
```

Comment:
BTW, the original purpose of this hack was so we could redact `RegistryAuth` when returning a service to the API user and avoid exposing the credentials, but I don't see this code doing that redaction right now. So we still have the hack, but not the thing that motivated it.